publication-commons/src/main/java/no/unit/nva/publication/service/impl/UpdateResourceService.java

```diff
@@ -98,30 +98,40 @@ public void updateOwner(SortableIdentifier identifier, UserInstance oldOwner, Us
         sendTransactionWriteRequest(request);
     }
     
-    PublishPublicationStatusResponse publishResource(UserInstance userInstance,
-                                                     SortableIdentifier resourceIdentifier)
+    PublishPublicationStatusResponse publishPublication(UserInstance userInstance,
+                                                        SortableIdentifier resourceIdentifier)
         throws ApiGatewayException {
-        List<Dao> daos = readResourceService
-                             .fetchResourceAndDoiRequestFromTheByResourceIndex(userInstance, resourceIdentifier);
-        var dao = extractResourceDao(daos);
-        var resource = (Resource) dao.getData();
-        if (resourceIsPublished(resource)) {
+        var publication = readResourceService.getPublication(userInstance, resourceIdentifier);
+        if (publicationIsPublished(publication)) {
             return publishCompletedStatus();
+        } else if (publicationIsDraft(publication)) {
+            publishPublication(publication);
+            return publishingInProgressStatus();
+        } else {
+            throw new UnsupportedOperationException("Functionality not specified");
         }
-        assertThatPublicationHasMinimumMandatoryFields(resource.toPublication());
-        setResourceStatusToPublished(daos, dao);
-        return publishingInProgressStatus();
+    }
+    
+    private static boolean publicationIsPublished(Publication publication) {
+        return PublicationStatus.PUBLISHED.equals(publication.getStatus());
+    }
+    
+    private void publishPublication(Publication publication) throws InvalidPublicationException {
+        assertThatPublicationHasMinimumMandatoryFields(publication);
+        publication.setStatus(PublicationStatus.PUBLISHED);
+        publication.setPublishedDate(clockForTimestamps.instant());
+        updatePublicationIncludingStatus(publication);
```

Comment:
the associatedArtifacts are changed when we are converting the publication to a resource. It will be addressed in the Resource Service refactoring